pkg/cli/alpha/scaffold.go

```diff
@@ -0,0 +1,257 @@
+				log.Fatal(err)
+			}
+		},
+	}
+	scaffoldCmd.Flags().StringVar(&projectConfig, "project-config", "",
+		"path to a kubebuilder project file if not in the current working directory")
+	scaffoldCmd.Flags().StringVar(&outputPath, "output", "",
+		"path to output the scaffolding. defaults to the current working directory")
+
+	return scaffoldCmd
+}
+
+func kubebuilderCreate(store store.Store) error {
+	resources, err := store.Config().GetResources()
+	if err != nil {
+		return err
+	}
+
+	var webhooks []resource.Resource
+	for i, r := range resources {
+		if r.Webhooks != nil {
+			// you cannot create a webhook without an API, create them after
+			webhooks = append(webhooks, resources[i])
+		} else {
+			if err = createAPI(r); err != nil {
+				return err
+			}
+		}
+	}
+
+	for _, w := range webhooks {
+		if err = createWebhook(w); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func createAPI(resource resource.Resource) error {
+	if resource.API == nil || resource.API.IsEmpty() {
+		return nil
+	}
+	var args []string
+	args = append(args, "create")
+	args = append(args, "api")
+	args = append(args, "--resource")
+	args = append(args, "--controller")
+	if resource.API.Namespaced {
+		args = append(args, "--namespaced")
+	}
+
+	args = append(args, getAPIGVK(resource)...)
+	return util.RunCmd("kubebuilder create api", "kubebuilder", args...)
+}
+
+func getAPIGVK(resource resource.Resource) []string {
+	var args []string
+	if len(resource.Plural) > 0 {
+		args = append(args, "--plural")
+		args = append(args, resource.Plural)
+	}
+
+	if len(resource.Group) > 0 {
+		args = append(args, "--group")
+		args = append(args, resource.Group)
+	}
+
+	if len(resource.Version) > 0 {
+		args = append(args, "--version")
+		args = append(args, resource.Version)
+	}
+
+	if len(resource.Kind) > 0 {
+		args = append(args, "--kind")
+		args = append(args, resource.Kind)
+	}
+	return args
+}
+
+func createWebhook(resource resource.Resource) error {
+	if resource.Webhooks == nil || resource.Webhooks.IsEmpty() {
+		return nil
+	}
+	var args []string
+	args = append(args, "create")
+	args = append(args, "webhook")
+
+	if resource.HasConversionWebhook() {
+		args = append(args, "--conversion")
+	}
+
+	if resource.HasValidationWebhook() {
+		args = append(args, "--programmatic-validation")
+	}
+
+	if resource.HasDefaultingWebhook() {
+		args = append(args, "--defaulting")
+	}
+
+	args = append(args, getAPIGVK(resource)...)
+	return util.RunCmd("kubebuilder create webhook", "kubebuilder", args...)
+}
+
+func kubebuilderEdit(store store.Store) error {
+	if store.Config().IsMultiGroup() {
+		args := []string{"edit", "--multigroup"}
+		return util.RunCmd("kubebuilder edit", "kubebuilder", args...)
+	}
+	return nil
+}
+
+func kubebuilderInit(store store.Store) error {
+	var args []string
+	args = append(args, "init")
+	args = append(args, generateInitArgs(store)...)
+	return util.RunCmd("kubebuilder init", "kubebuilder", args...)
```

Comment:
We need to check all plugins used to initiate the project and use them if we found them tracked in the PROJECT file.
Also, if the plugin has data tracked we need to use them to pass the flags with the respective values.

See for example the case of we use the deploy image p;lugin.
You can use the samples under the test data to check it out.

Reply:
`generateInitArgs` does loop over the plugin chain. So you are saying supply the `options` to the plugin as well?

Reply:
Note that when we initiate a project we can use plugins and indeed its flags. (i.e `kubebuilder init --plugins="go/v4-alpha,declarative,grafana/v1-alpha"`)

Also, when we run create API we can also use a plugin and pass values/flags (i.e.  ` kubebuilder create api --group example.com --version v1alpha1 --kind Memcached --image=memcached:1.4.36-alpine --image-container-command="memcached,-m=64,-o,modern,-v" --image-container-port="11211" --run-as-user="1001" --plugins="deploy-image/v1-alpha"`  )

So when we re-scaffold we need to check in the PROJECT file the plugins 
a) used in the layout, see: https://github.com/kubernetes-sigs/kubebuilder/blob/8bd569de6745d94249a526319f921bbb4f776bb2/testdata/project-v3-addon-and-grafana/PROJECT#L2-L5
b) used to create the apis, see: 
- i.e with declarative: plugins:
  declarative.go.kubebuilder.io/v1:
    resources:
    - domain: testproject.org
      group: crew
      kind: Captain
      version: v1
    - domain: testproject.org
      group: crew
      kind: FirstMate
      version: v1
    - domain: testproject.org
      group: crew
      kind: Admiral
      version: v1
  grafana.kubebuilder.io/v1-alpha: {}

- i.e. with deploy image: https://github.com/kubernetes-sigs/kubebuilder/blob/8bd569de6745d94249a526319f921bbb4f776bb2/testdata/project-v3-with-deploy-image/PROJECT#L4-L21

Reply:
But we can try to make it as simple as possible. If the implementation is able to work with the deploy image, (see the sample https://github.com/kubernetes-sigs/kubebuilder/tree/master/testdata/project-v3-with-deploy-image generate ) then, it can work in the biggest part of the use cases.